script2.js

```diff
@@ -0,0 +1,121 @@
+var win = 0;
+var loss = 0;
+var draw = 0;
+var username = "";
+var waitingForUserName = true;
+var gameMode = "normal";
+
+var main = function (input) {
+  var myOutputValue = 'Scissors Paper Stone!';
+  var comChoice = computerChoice();
+
+  if (waitingForUserName)
+  {
+      username = input;
+      waitingForUserName = false;
+      myOutputValue = "Welcome " + username;
+      return myOutputValue;
+  }
+  else
+  {
+    if (input != "Scissors" && input != "Paper" && input != "Stone" && input != "reverse" && input != "normal")
+    {
+      myOutputValue = "Please input a valid choice.";
+      return myOutputValue;
+    }
+
+    if (input == "reverse")
+    {
+      gameMode = "reverse";
+      return "Game mode has been reversed!";
+    }
+    else if (input == "normal")
+    {
+      gameMode = "normal";
+      return "Game mode is normal.";
+    }
+
+    if (gameMode == "normal")
+    {
+      if (input == comChoice)
+      {
+        myOutputValue = "You chose: " + input;
+        myOutputValue += "<br>Computer chose: " + comChoice;
+        myOutputValue += "<br>DRAW!<br>";
+        draw++;
+        myOutputValue += "<br>Congratulations " + username + ", Wins: " + win + " " + "Losses: " + loss + " " + "Draws: " + draw;
+      }
+      else if ((input == "Scissors" && comChoice == "Paper") || (input == "Paper" && comChoice == "Stone") || input == "Stone" && comChoice == "Scissors")
+      {
+        myOutputValue = "You chose: " + input;
+        myOutputValue += "<br>Computer chose: " + comChoice;
+        myOutputValue += "<br>WIN!!<br>";
```

Comment:
You could combine all of these statements into one. Generally, I recommend using template literals ( ` ) instead of quotations as it allows you to write out these statements without having to use + between vars and strings. It would something like this -

myOutputValue = ` You chose: ${input} \n Computer chose: ${comChoice} \n WIN!! \n`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
I tried using template literals for some statements in the Beat That assignment, but not so familiar with it.. would need more practice and testing